### PR TITLE
ref(quick-start): Add analytics to track group completion

### PR DESCRIPTION
**Pull request by priscilawebdev**
closes https://github.com/getsentry/projects/issues/331

### CODE REVIEW

File: static/app/components/sidebar/newOnboardingStatus.tsx [hunk 1]
```diff
@@ -96,7 +97,7 @@ export function NewOnboardingStatus({
   }, [onShowPanel, isActive, walkthrough, markDoneTaskAsComplete, organization]);
 
   useEffect(() => {
-    if (totalRemainingTasks !== 0 || isActive) {
+    if (totalRemainingTasks !== 0 || hasQuickStartBeenCompleted.current) {
```
Comment (priscilawebdev) [inline static/app/components/sidebar/newOnboardingStatus.tsx:100]: We can trigger the analytics to track whether all groups have been completed before closing the sidebar. This change does that

File: static/app/components/onboardingWizard/newSidebar.tsx [hunk 2]
```diff
@@ -212,14 +217,31 @@ function TaskGroup({
...
+      previousCompletedTasksCount !== tasks.length - 1
+    ) {
+      return;
+    }
+
+    trackAnalytics('quick_start.task_group_completed', {
+      organization,
+      group,
+    });
+  }, [previousCompletedTasksCount, completedTasks, tasks, group, organization]);
```
Comment (priscilawebdev) [inline static/app/components/onboardingWizard/newSidebar.tsx:243]: this is the logic to track whether a group has been completed

File: static/app/utils/analytics/quickStartAnalyticsEvents.tsx [hunk 3]
```diff
@@ -12,10 +12,14 @@ export type QuickStartEventParameters = {
     todo_id: string;
     todo_title: string;
   };
+  'quick_start.task_group_completed': {
+    group: 'getting_started' | 'beyond_basics';
```
Comment (priscilawebdev) [inline static/app/utils/analytics/quickStartAnalyticsEvents.tsx:16]: we do not need the 'new_experience' here as it is present only in the new experience

File: static/app/components/onboardingWizard/newSidebar.tsx [hunk 4]
```diff
@@ -212,14 +217,31 @@ function TaskGroup({
...
+  const previousCompletedTasksCount = usePrevious(completedTasks.length);
 
   useEffect(() => {
     setIsExpanded(expanded);
   }, [expanded]);
 
+  useEffect(() => {
+    if (
+      completedTasks.length !== tasks.length ||
+      previousCompletedTasksCount !== tasks.length - 1
```
Comment (ArthurKnaus) [inline static/app/components/onboardingWizard/newSidebar.tsx:234]: What if multiple tasks get marked as done at once?
Instead we could check `previousCompletedTaskCount === task.length`, right?

File: static/app/components/sidebar/newOnboardingStatus.tsx (hunk 1 above)
Comment (ArthurKnaus) [reply static/app/components/sidebar/newOnboardingStatus.tsx:100]: Those analytics will fire on every component mount once all tasks are done, is this intentional?

Comment (priscilawebdev) [reply static/app/components/sidebar/newOnboardingStatus.tsx]: oh I missed that! good catch 🙏 this commit solves the issue https://github.com/getsentry/sentry/pull/79911/commits/0ea80e62c47861c53fc313a9a77329f27d847c72

File: static/app/components/onboardingWizard/newSidebar.tsx (hunk 4 above)
Comment (priscilawebdev) [reply static/app/components/onboardingWizard/newSidebar.tsx]: no it still does not solve the issue. To solve this I decided to use local storage

File: static/app/components/onboardingWizard/newSidebar.tsx [hunk 5]
```diff
@@ -378,9 +410,12 @@ export function NewOnboardingSidebar({
             tasks={sortedBeyondBasicsTasks}
             hidePanel={onClose}
             expanded={
-              groupTasksByCompletion(gettingStartedTasks).incompletedTasks.length === 0
+              groupTasksByCompletion(sortedGettingStartedTasks).incompletedTasks
+                .length === 0 &&
+              groupTasksByCompletion(sortedBeyondBasicsTasks).incompletedTasks.length > 0
```
Comment (priscilawebdev) [inline static/app/components/onboardingWizard/newSidebar.tsx:415]: minor issue I fixed while testing the analytics. hope that is fine

**Review by ArthurKnaus — APPROVED** [review]

**Pull request merged**